pkg/process/manager.go

```diff
@@ -191,6 +203,57 @@ type Opt struct {
 	Database         *store.Database
 	DaemonMode       string
 	CacheDir         string
+	RecoverPolicy    string
+}
+
+func (m *Manager) handleDaemonDeathEvent() {
+	for event := range m.LivenessNotifier {
+		log.L.Warnf("Daemon %s died! socket path %s", event.daemonID, event.path)
+
+		if m.RecoverPolicy == "restart" {
+			// We can't restart nydusd for now fuse connection
+			d := m.GetByDaemonID(event.daemonID)
+
+			if err := waitDaemon(d); err != nil {
+				log.L.Warnf("fails to wait for daemon")
+			}
+
+			// Starting a new nydusd will re-subscribe
+			if err := m.monitor.Unsubscribe(d.ID); err != nil {
+				log.L.Warnf("fails to unsubscribe")
+			}
+
+			clearDaemonVestige(d)
+
+			m.StartDaemon(d)
+
+			if d.IsSharedDaemon() {
+				daemons := m.daemonStates.List()
+				for _, d := range daemons {
+					if d.ID != daemon.SharedNydusDaemonID {
+						if err := d.SharedMount(); err != nil {
+							log.L.Warnf("fail to mount rafs, %v", err)
+						}
+					}
+				}
+			}
+		} else if m.RecoverPolicy == "failover" {
+			log.L.Warnf("failover is implemented now!")
+		}
+	}
+}
+
+func clearDaemonVestige(d *daemon.Daemon) {
```

Comment:
Better add it as a method `CleanMount` of `Daemon` struct.

Reply:
Ok, maybe `d.ClearVestige`

Reply:
fixed